src/logic/operate.js

```diff
@@ -0,0 +1,45 @@
+import Big from 'big.js';
+
+const operate = (numberOne, numberTwo, operation) => {
+  let total;
+  let result;
+
+  if (operation === '+') {
+    total = Big(numberOne).plus(numberTwo).toString();
+    result = { total: null, next: total, operation };
+  }
+
+  if (operation === '-') {
+    total = Big(numberOne).minus(numberTwo).toString();
+    result = { total: null, next: total, operation };
+  }
+
+  if (operation === '÷') {
+    if (numberTwo === '0') {
+      result = { total: 'Math Error', next: null, operation };
+    } else {
+      total = Big(numberOne).div(numberTwo).toString();
+      result = { total: null, next: total, operation };
+    }
+  }
+
+  if (operation === 'X') {
+    total = Big(numberOne).times(numberTwo).toString();
+    result = { total: null, next: total, operation };
+  }
+
+  if (operation === '%') {
+    if (numberOne === null) {
+      total = Big(numberOne).div(100).toString();
+      result = { total, next: null, operation };
+    }
+    if (numberTwo === null && numberOne != null) {
+      total = Big(numberTwo).div(100).toString();
+      result = { total, next: null, operation };
+    }
+  }
```

Comment:
Instead of continuously repeating the `Big(numberOne)` and `Big(numberTwo)` expressions, you can create 2 variables to store the `Big` version of `numberOne` and `numberTwo` at the beginning of the  function. e.g.
```
const bigOne =  Big(numberOne);
const bigTwo =  Big(numberTwo);
```
You can therefore reuse the `bigOne` and `bigTwo` variables throughout the function.

You can also use `mod` for modulus operation. e.g.
```
  if (operation === '%') {
    return bigOne.mod(bigTwo).toString();
  }

```